docs/DeveloperGuide.md

```diff
@@ -2,37 +2,615 @@
+`Parser` to extract the relevant keywords to prepare the appropriate `Command` object. It is also reponsible for
+handling invalid inputs by the user. The `Parser` consist of the following methods.
+
+- `parse()`: Extracts out the command keyword from the user input.
+- `prepareCommand()`: Prepares the appropriate `Command` object based on the commmand keyword and the other
+relevant input fields provided by the user. It also checks if the format of the command is correct.
+- `isCorrectMarker()`: Checks if the marker used to label the input fields are correct.
+- `isValidImportance()`: Checks if the importance value provided by the user is a valid integer and within the [1, 10]
+range.
+- `isValidKeyword()`: Checks if the `keyWord` provided by the user is valid when preparing the `FilterCommand`.
+- `isValidCommand()`: Checks if the command keyword provided by the user is valid.
+
+Given below is an example of how the `Parser` works when it is issued a remove command.
+
+> **Step 1**: The user enters a command. The full command is read in by the `Ui`. `CLIAlgo` invokes the `parse()`
+> method from the `Parser`.
+
+> **Step 2**: The `parse()` method extracts out the command keyword provided by the user. It then calls the 
+> `prepareCommand()` method.
+
+> **Step 3**: The `parepareCommand()` identifies the correct `Command` object to prepare based on the command keyword.
+> Since the command keyword provided is `remove`, `prepareCommand()` calls `prepareRemoveCommand()`.
+
+> **Step 4**: If the `NAME` field of the command is null or not labelled using the correct marker, 
+> `prepareRemoveCommand()` returns an `InvalidCommand` object. If the name of the `CS2040CFile` provided is not
+> found, it returns a `NameNotFoundCommand`. If the `NAME` input field is valid, a `RemoveCommand` object is returned.
+
+The following sequence diagram shows how the Parser work.
+
+![](.\\sequence\\diagrams\\Parser.png "Parser Sequence Diagram")
+
+### Initializing previous saved data feature
+
+#### Current implementation
+
+The function for reading the previously saved data is facilitated by the `FileManager`. The `FileManager`
+creates a `SingleFile` for each valid topic name and invokes `createNewFile` for those files in the for
+in `TOPIC_NAME.txt` in the folder `.\\data`. If the files already exist, they are not created. Instead,
+the contents of the file would be read line-by-line. The read data would then be passed to `FileDecoder`
+which would then convert these raw data into `CS2040CFile` objects. The `CS2040File` objects are then passed
+into a `HashMap` which represents the topic these `CS2040CFile` objects belong to. The `HashMap` is then passed
+back to the `TopicManager`, completing the initialization process.
+
+The following sequence diagram shows how previously saved files are loaded into `CLIAlgo`.
+
+![](.\\sequence\\diagrams\\InitializationFileManager.png "FileManager Initialization Sequence Diagram")
```

Comment:
<img width="443" alt="image" src="https://user-images.githubusercontent.com/88618401/228723082-da5f1177-f79f-4518-9b12-6126a07e5576.png">
i think the activation bar inside the alt block should not be broken into two parts